ui-speedspacechart/src/components/layers/SpeedLimitTagsLayer.tsx

```diff
@@ -18,12 +18,44 @@ const SpeedLimitTagsLayer = ({ width, marginTop, store }: SpeedLimitTagsLayerPro
   const tooltip = useRef<tooltipInfos | null>();
 
   useEffect(() => {
-    const updateTooltip = async () => {
+    const updateCanvas = async () => {
       const currentCanvas = canvas.current as HTMLCanvasElement;
       const ctx = currentCanvas.getContext('2d') as CanvasRenderingContext2D;
-      tooltip.current = await drawSpeedLimitTags({ ctx, width, height: marginTop, store });
+      const restrictedStore = {
+        speedLimitTags: store.speedLimitTags,
+        ratioX: store.ratioX,
+        leftOffset: store.leftOffset,
+        layersDisplay: {
+          electricalProfiles: store.layersDisplay.electricalProfiles,
+          powerRestrictions: store.layersDisplay.powerRestrictions,
+        },
+        speeds: store.speeds,
+      };
+      await drawSpeedLimitTags({
+        ctx,
+        width,
+        height: marginTop,
+        store: restrictedStore as RestrictedStore,
```

Comment:
I prefer to avoid `as` when possible, because it basically turns off the type checks. I don't think it's necessary here, `const restrictedStore: RestrictedStore = …` should work fine I believe (or just leave it up to type inference, should work too).

Reply:
Ok, will do!

Reply:
Ok so I hadn't constrained my type correctly with Pick, that's fixed